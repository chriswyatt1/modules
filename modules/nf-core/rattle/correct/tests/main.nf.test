nextflow_process {

    name "Test Process RATTLE_CORRECT"
    script "../main.nf"
    process "RATTLE_CORRECT"

    tag "modules"
    tag "modules_nfcore"
    tag "rattle"
    tag "rattle/correct"

    test("test-rattle") {

        when {
            process {
                """
                input[0] = [
				    [ id:'test', single_end:false ], // meta map
				    [ file('${projectDir}/sample.fastq', checkIfExists: true) ]
				]
                input[1] = [
				    [ id:'test', single_end:false ], // meta map
				    [ file('${projectDir}/clusters.out', checkIfExists: true) ]
				]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("test-rattle-stub") {
        options '-stub'
        when {
            process {
                """
                input[0] = [
				    [ id:'test', single_end:false ], // meta map
				    [ file('${projectDir}/sample.fastq', checkIfExists: true) ]
				]
                input[1] = [
				    [ id:'test', single_end:false ], // meta map
				    [ file('${projectDir}/clusters.out', checkIfExists: true) ]
				]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

}
