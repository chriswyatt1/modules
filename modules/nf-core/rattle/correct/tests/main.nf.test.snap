{
    "test-rattle": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    
                ],
                "2": [
                    
                ],
                "3": [
                    
                ],
                "consensi": [
                    
                ],
                "corrected": [
                    
                ],
                "uncorrected": [
                    
                ],
                "versions": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-02-06T15:46:21.066136548"
    }
}